zk_prover/src/merkle_sum_tree/tests.rs

```diff
@@ -87,6 +87,85 @@ mod test {
         proof_invalid_3.sibling_sums[0] = [0.into(), 0.into()];
     }
 
+    #[test]
+    fn test_update_mst_leaf() {
```

Comment:
Can you create a test (or modify this one) that: 

- Build a first merkle tree from `entry_16`
- Update leaf 7 of the merkle tree with balances 2087, 79731
- Build a second merkle tree from `entry_16_updated` that has the 7th username with balances 2087, 79731
- Check that the merkle root and the merkle balances of the first updated tree and the second tree are matching